AIPscan/Reporter/templates/report_format_versions_count.html

```diff
@@ -0,0 +1,51 @@
+{% extends "report_base.html" %}
+
+{% block content %}
+
+  <div class="alert alert-secondary">
+    <a class="noprint" onClick="window.print();return false">
+      <button type="button" class="btn btn-info" style="float:right;">Print</button>
+    </a>
+    <strong>Report:</strong> File format version count
+    <br>
+    <strong>Storage Service:</strong> {{ storage_service_name }}
+    <br>
+    {% if start_date and end_date %}
+      {{ total_file_count }} original files ingested between {{ start_date.strftime('%Y-%m-%d') }} and {{ end_date.strftime('%Y-%m-%d') }}
+    {% elif start_date %}
+      {{ total_file_count }} original files ingested on and since {{ start_date.strftime('%Y-%m-%d') }}
+    {% elif end_date %}
+      {{ total_file_count }} original files ingested on and before {{ end_date.strftime('%Y-%m-%d') }}
+    {% else %}
+      {{ total_file_count }} original files
+    {% endif %}
+    <br>
+    {{ puid_count }} different format versions totalling {{ total_size | filesizeformat }}
+  </div>
+
+  {% if versions %}
+    <table id="formatversioncounts" class="table table-striped table-bordered">
+      <thead>
+        <tr>
+          {% for column in columns %}
+          <th><strong>{{ column }}</strong></th>
+          {% endfor %}
+        </tr>
+      </thead>
+      <tbody>
+      {% for version in versions %}
+        <tr>
+          <td>{{ version["PUID"] }}</td>
+          <td>{{ version["Format"] }}</td>
+          <td>{{ version["Version"] }}</td>
+          <td>{{ version["Count"] }}</td>
+          <td>{{ version["Size"] | filesizeformat }}</td>
```

Comment:
This might be related to the reporter module code, but here, I have a `{'PUID': 'fmt/468', 'Format': 'ISO Disk Image File', 'Version': None, 'Count': 1, 'Size': None}` result and the template expects a number to work with and not `None` so I wonder how we can fortify this or the data coming out of the DB (or into) which can help us to avoid this? I looked at the original METS and it looks like the `ISO Disk Image File` was ingested as a standard transfer from one of my FPR test sets - but it's likely the image from the sample data dirs.

Reply:
I think it's likely that this `File` object has the format and size it does because of this potentially misleading `AttributeError` handling: https://github.com/artefactual-labs/AIPscan/blob/39dbf8801e0b52e58f6e3f1cc9f438929849b85e/AIPscan/Aggregator/database_helpers.py#L196-L200 (see also issue #83). I wonder if it's actually an ISO disk image...

Regardless, I think you hit the nail on the head that we should be checking for `None` in the data endpoint to avoid this issue. Good spot!

Reply:
Ah whoops, just re-read your comment and my "I wonder if it's actually an ISO disk image..." was off base here. Sorry about that! :sweat_smile: